src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,638 @@
+                //     void M(ref readonly ref readonly int p) { }
+                Diagnostic(ErrorCode.ERR_DupParamMod, "readonly").WithArguments("readonly").WithLocation(3, 29)
+            };
+
+            CreateCompilation(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+            CreateCompilation(source).VerifyDiagnostics(expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ReadonlyWithoutRef()
+        {
+            var source = """
+                class C
+                {
+                    void M(readonly int p) => throw null;
+                }
+                """;
+            var expectedDiagnostics = new[]
+            {
+                // (3,12): error CS9501: 'readonly' modifier must be specified after 'ref'.
+                //     void M(readonly int p) => throw null;
+                Diagnostic(ErrorCode.ERR_RefReadOnlyWrongOrdering, "readonly").WithLocation(3, 12)
+            };
+
+            CreateCompilation(source, parseOptions: TestOptions.Regular11).VerifyDiagnostics(expectedDiagnostics);
+            CreateCompilation(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+            CreateCompilation(source).VerifyDiagnostics(expectedDiagnostics);
+        }
+
+        [Fact]
+        public void ReadonlyWithParams()
+        {
+            var source = """
+                class C
+                {
+                    void M(readonly params int[] p) => throw null;
+                }
+                """;
+            var expectedDiagnostics = new[]
+            {
+                // (3,12): error CS9501: 'readonly' modifier must be specified after 'ref'.
+                //     void M(readonly params int[] p) => throw null;
+                Diagnostic(ErrorCode.ERR_RefReadOnlyWrongOrdering, "readonly").WithLocation(3, 12)
+            };
+
+            CreateCompilation(source, parseOptions: TestOptions.Regular11).VerifyDiagnostics(expectedDiagnostics);
+            CreateCompilation(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+            CreateCompilation(source).VerifyDiagnostics(expectedDiagnostics);
+        }
+
+        [Fact]
+        public void RefReadonlyWithParams_01()
+        {
+            var source = """
+                class C
+                {
+                    void M(params ref readonly int[] p) => throw null;
+                }
+                """;
+            var expectedDiagnostics = new[]
+            {
+                // (3,19): error CS1611: The params parameter cannot be declared as ref
+                //     void M(params ref readonly int[] p) => throw null;
+                Diagnostic(ErrorCode.ERR_ParamsCantBeWithModifier, "ref").WithArguments("ref").WithLocation(3, 19),
+                // (3,23): error CS9501: 'readonly' modifier must be specified after 'ref'.
```

Comment:
>// (3,23): error CS9501: 'readonly' modifier must be specified after 'ref'. [](http://example.com/codeflow?start=16&length=75)

This error feels misleading, `readonly` is specified after `ref` in code. I think we should suppress this error.